### PR TITLE
Misc doc tweaks

### PR DESCRIPTION
**Pull request by Dieterbe**

### CODE REVIEW

File: docs/CONTRIBUTING.md [hunk 1]
```diff
@@ -21,6 +21,7 @@ see the [Makefile](../Makefile) for more targets
 1. functions, methods, types should be clearly understandable, either through an obvious name, or documentation when needed.
 2. all code documentation must conform to [golang best practices](https://blog.golang.org/godoc-documenting-go-code)
 3. add unit tests for tricky or non-trivial code. Not needed for obvious, simple or glue code.  Use your best judgment or just ask us.
+   In particular, for bugfixes, try to commit first the test and than the patch that fixes the test.
```
Comment (replay) [inline docs/CONTRIBUTING.md]: I usually fix the bug fix, commit the change, then write  a unit test to verify it's fixed, then switch the order of the two commits, then verify that without the fix commit the test fails and with the fix commit it succeeds. but anyway, in the git history / PRs that results in the same...

Comment (shanson7) [inline docs/CONTRIBUTING.md]: "and then"

Comment (Dieterbe) [reply docs/CONTRIBUTING.md]: i often do the same. good point the ordering is worded as temporal rather than revision-history wise

Comment (robert-milan) [inline docs/CONTRIBUTING.md]:
```suggestion
   In particular, for bugfixes, try to first commit the test and then the patch that fixes the test.
```

**Pull request merged**